src/axom/sidre/tests/spio/spio_parallel.hpp

```diff
@@ -92,6 +95,10 @@ TEST(spio_parallel, parallel_writeread)
     i1_vals[i] = (i + 10) * (404 - my_rank - i);
   }
 
+  // Add an array view of size 0 under group "c"
+  Group* gc = flds2->createGroup("c");
+  gc->createView("i2")->allocate(DataType::c_int(0));
```

Comment:
I found something!  
Adding a zero-sized array to the `spio_parallel` test generates the following warning:

```
188: [ RUN      ] spio_parallel.parallel_writeread
188: 
188: ***********************************
188: [WARNING in line 198 of file <axom>/src/axom/sidre/core/Buffer.cpp]
188: MESSAGE=Failed Check: src != nullptr
188: Cannot copy data into Buffer from null pointer.
188: 
188: ***********************************
188: [       OK ] spio_parallel.parallel_writeread (125 ms)
```

So, the issue might be related to how we write/read these arrays to `hdf5` through `spio`.

Reply:
This was not the problem!

I found and fixed an overeager warning when importing zero-sized arrays from `hdf5` to `sidre` via `conduit`.
But, the code was already doing the right thing, it was just adding an extra warning.